src/controls/framework/controlPropertyEditor.ts

```diff
@@ -0,0 +1,53 @@
+import { ControlContainer } from "../components/design/control-container.e";
+import { ComponentChild, render } from "@friendly/elements/jsxElements";
+import { setPropertyUndoRedo } from "../controls/editors/_shared";
+
+interface JsxEditorRefreshArguments<T> {
+  old: T;
+  new: T;
+  canMerge?: boolean;
+}
+
+export interface IPropertyEditor {
+  elementToMount: HTMLElement;
+}
+
+/**
+ * Creates an editor that uses JSX to provide the contents.
+ * @param instance the instance for which the editor is valid
+ * @param callback a callback that can be used to re-render the editor
+ * @returns an IPropertyEditor that edits the given property
+ */
+export function createJsxEditor<T>(
```

Comment:
Note that none of this was working as part of this prototype - we thought to separate editors from properties, but ditched the entire thing before doing so